Formula/cpplint.rb

```diff
@@ -0,0 +1,24 @@
+class Cpplint < Formula
+  include Language::Python::Virtualenv
+
+  desc "Static code checker for C++"
+  homepage "https://github.com/cpplint/cpplint"
+  url "https://github.com/cpplint/cpplint/archive/1.5.4.tar.gz"
+  sha256 "e254b5620fb039689b8f5e64b07384ee53beb695304c5c01195133be662b4457"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+  end
+
+  depends_on "python@3.8"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    output = shell_output("#{bin}/cpplint --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.